tests/unit/entrypoint/database/test_topic_filter.py

```diff
@@ -0,0 +1,323 @@
+# Blue Brain Search is a text mining toolbox focused on scientific use cases.
+#
+# Copyright (C) 2020  Blue Brain Project, EPFL.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import argparse
+import inspect
+import pathlib
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from bluesearch.database.topic_info import TopicInfo
+from bluesearch.database.topic_rule import TopicRule
+from bluesearch.entrypoint.database import topic_filter
+from bluesearch.entrypoint.database.topic_filter import (
+    filter_topics,
+    parse_filter_config,
+)
+from bluesearch.utils import JSONL
+
+TOPIC_FILTER_PARAMS = {
+    "extracted_topics",
+    "filter_config",
+    "output_file",
+}
+
+
+def test_init_parser():
+    parser = topic_filter.init_parser(argparse.ArgumentParser())
+
+    args = parser.parse_args(["/path/to/topics", "/path/to/config", "/path/to/output"])
+    assert vars(args).keys() == TOPIC_FILTER_PARAMS
+
+    # Test the values
+    assert args.extracted_topics == pathlib.Path("/path/to/topics")
+    assert args.filter_config == pathlib.Path("/path/to/config")
+    assert args.output_file == pathlib.Path("/path/to/output")
+
+
+def test_run_arguments():
+    assert inspect.signature(topic_filter.run).parameters.keys() == TOPIC_FILTER_PARAMS
+
+
+EXTRACTED_TOPICS_EXAMPLE = [
```

Comment:
Maybe move to `tests/data/` to avoid having this long text in the test?

Reply:
For sure possible. The main reason why I decided to keep it inside of the source code is that after dumping to `.jsonl` there will be 1 topic info (a big JSON) per line. Arguably, that makes it less readable unless processed somehow in your text editor.

Reply:
Yeah, sounds like a good reason.